plugin/confirm_quit.vim

```diff
@@ -105,7 +105,7 @@ fun! s:going_to_quit()
 		endif
 	endfor
 
-	return l:open_bufs == 1
+	return l:open_bufs == 1 && winnr('$') == 1
```

Comment:
Right now we get a list of all open buffers, and quit if only one buffer is open.

This doesn't work as expected because you can have more than one view on a buffer (which is what a window is).

Wouldn't it make more sense to just get the number of open windows, instead (which is what `winnr('$')` does)? Not sure why it needs to check the number of loaded buffers at all. The reason I wrote it like this is probably because I don't use splits/windows, so don't think about it much (... but it's been 4 years since I wrote it, so not sure).

Right now this will also fail with tabs (even with your patch):

    :e file
    :tabedit file
    :q

I think this entire function could possibly be replaced with:

    fun! s:going_to_quit()
        return tabpagenr('$') is 1 && winnr('$') is 1
    endfun

But I'm not entirely sure if that covers everything :thinking: Seems to work well in a quick test anyway.

Reply:
I agree that checking the number of tabs and windows seems like a better solution.
I use mainly windows myself so it didn't occur to me to account for tabs in the original fix.

Reply:
Cheers, thanks!